base/abstractdict.jl

```diff
@@ -130,6 +140,18 @@ values(a::AbstractDict) = ValueIterator(a)
 Return an iterator over `key => value` pairs for any
 collection that maps a set of keys to a set of values.
 This includes arrays, where the keys are the array indices.
+
+# Implementation
+
+It is highly recommended that each implementation of `pairs` satisfies
+the following relationships:
+
+```julia
+isequal(map(first, paris(collection)), keys(collection))
+isequal(map(last, paris(collection)), values(collection))
```

Comment:
Note, that currently neither `Array` nor `Dict` satisfies these relationships, because `map` is not defined for the return types of `pairs`:
```
julia> coll = [10, 20];
julia> map(first, pairs(coll))
ERROR: map is not defined on dictionaries
```